parse_test.go

```diff
@@ -860,6 +860,26 @@ func TestEnvironmentVariableInSubcommandIgnored(t *testing.T) {
 	assert.Equal(t, "", args.Sub.Foo)
 }
 
+func TestParserMustParse(t *testing.T) {
+	originalExit := osExit
+	originalStdout := stdout
+	defer func() {
+		osExit = originalExit
+		stdout = originalStdout
+	}()
+
+	var exitCode *int
+	osExit = func(code int) { exitCode = &code }
+	stdout = &bytes.Buffer{}
+
+	p, err := NewParser(Config{}, &struct{}{})
+	require.NoError(t, err)
+	assert.NotNil(t, p)
+
+	p.MustParse(nil)
+	assert.Equal(t, 0, *exitCode)
```

Comment:
Yup, added it. This test fails though and it doesn't feel like it should. The `exitCode` remains nil.

Reply:
I believe that's because when you pass `nil` to `MustParse` you are essentially parsing an empty list of command line arguments into an empty struct, just as if you did

```go
var args struct {
}
arg.MustParse(&args)
```
and then invoked the program with
```
./program
```
Since there are no invalid or missing arguments, go-arg says that this is perfectly acceptable.

Reply:
Shouldn't that still result in an exit code and one of 0?

Reply:
In the case that there are no errors, `MustParse` does not exit, since otherwise it would *always* exit and nobody would have a chance to process their command line arguments even if parsing succeeded. The `osExit = func...` in your test overrides the function that `MustParse` calls when trying to exit, but that function doesn't get called at all if parsing succeeds.